src/main/java/su/litvak/chromecast/api/v2/Channel.java

```diff
@@ -124,56 +124,32 @@ public void run() {
+                        jsonMSG = message.getPayloadUtf8().replaceFirst("\"type\"", "\"responseType\"");
                         if (jsonMSG == null || jsonMSG.isEmpty()) {
                             LOG.warn(" <-- Received empty message. Ignore.");
                             continue;
                         }
 
                         // Determine whether the message belongs to cast protocol or is a custom
                         // message from the receiver app
-                        final JsonNode parsed;
-                        try {
-                            parsed = jsonMapper.readTree(jsonMSG);
-                        } catch (JsonProcessingException jpex) {
-                            // Ignore
-                            // To prevent potential null pointer access using parsed, we could continue...
-                            continue;
-                        }
-
-                        if (isAppEvent(parsed)) {
-                            AppEvent event = new AppEvent(
-                                    message.getNamespace(), message.getPayloadUtf8());
-                            notifyListenersAppEvent(event);
-                        } else {
-                            if (parsed.has("requestId")) {
-                                Long requestId = parsed.get("requestId").asLong();
-                                final ResultProcessor<? extends Response> rp = requests.remove(requestId);
-                                if (rp != null) {
-                                    rp.put(jsonMSG);
-                                } else {
-                                    notifyListenersOfSpontaneousEvent(parsed);
-                                }
-                            } else if (parsed.has("responseType") && parsed.get("responseType").asText().equals("MEDIA_STATUS")) {
-                                notifyListenersOfSpontaneousEvent(parsed);
-                            } else if (parsed.has("responseType") && parsed.get("responseType").asText().equals("PING")) {
-                                write("urn:x-cast:com.google.cast.tp.heartbeat", StandardMessage.pong(), DEFAULT_RECEIVER_ID);
-                            } else if (parsed.has("responseType") && parsed.get("responseType").asText().equals("CLOSE")) {
-                                notifyListenersOfSpontaneousEvent(parsed);
-                            }
-                        }
+                        parsed = jsonMapper.readTree(jsonMSG);
                     } else {
                         LOG.warn("Received unexpected {} message", message.getPayloadType());
                     }
                 } catch (InvalidProtocolBufferException ipbe) {
-                    LOG.debug("Error while processing protobuf: {}", ipbe.getLocalizedMessage());
+                    LOG.warn("Error while processing protobuf: {}", ipbe);
```

Comment:
you're totally right. it actually won't print anything, but it's poor form. updating...